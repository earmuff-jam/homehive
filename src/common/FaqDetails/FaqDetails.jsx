import React from "react";

import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Box,
  IconButton,
  Stack,
  Typography,
} from "@mui/material";
import RowHeader from "common/RowHeader/RowHeader";

export default function FaqDetails({ data }) {
  return (
    <Stack spacing={2} alignItems="center">
      <RowHeader
        title="Frequently asked questions"
        caption="Answers to common questions you may have."
      />

      <Box sx={{ mx: "auto", maxWidth: "50%" }}>
        {data?.map((item, index) => {
          return (
            <Accordion
              key={index}
              defaultExpanded={index === 0}
              disableGutters
              elevation={0}
              sx={{ marginBottom: "1rem" }}
            >
              <AccordionSummary
                expandIcon={<ExpandMoreIcon fontSize="small" />}
              >
                <Stack direction="row" spacing={1}>
                  <IconButton size="small">{item.icon}</IconButton>
                  <Typography fontWeight="bold">{item.q}</Typography>
                </Stack>
              </AccordionSummary>
              <AccordionDetails>
                <Typography variant="body2" sx={{ color: "text.secondary" }}>
                  {item.ans}
                </Typography>
              </AccordionDetails>
            </Accordion>
          );
        })}
      </Box>
    </Stack>
  );
}
