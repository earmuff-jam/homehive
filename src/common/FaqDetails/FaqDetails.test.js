import React from "react";

import FaqDetails from "./FaqDetails";
import { HelpOutline } from "@mui/icons-material";
import { fireEvent, render, screen } from "@testing-library/react";

describe("FaqDetails component tests", () => {
  describe("it should render and pass all snapshot tests", () => {
    const { asFragment } = render(<FaqDetails />);
    expect(asFragment()).toMatchSnapshot();
  });
  describe("it should render and pass all component tests", () => {
    const mockData = [
      {
        q: "What is Invoicer?",
        ans: "Invoicer is a billing app for landlords.",
        icon: <HelpOutline data-testid="faq-icon-1" />,
      },
      {
        q: "How do I add a tenant?",
        ans: "Go to the tenants page and click add tenant.",
        icon: <HelpOutline data-testid="faq-icon-2" />,
      },
    ];

    // check the first render here because of how we are spreading the
    // sxProps on RowHeader component, mui tries to copy node elements during test.
    test("renders the FAQ header and questions", () => {
      render(<FaqDetails data={mockData} />);

      expect(
        screen.getAllByText("Frequently asked questions")[0],
      ).toBeInTheDocument();
      expect(
        screen.getAllByText("Answers to common questions you may have.")[0],
      ).toBeInTheDocument();

      expect(screen.getByText("What is Invoicer?")).toBeInTheDocument();
      expect(screen.getByText("How do I add a tenant?")).toBeInTheDocument();
    });

    test("expands and collapses accordion", () => {
      render(<FaqDetails data={mockData} />);

      const firstAnswer = screen.getByText(
        "Invoicer is a billing app for landlords.",
      );
      // First one is expanded by default
      expect(firstAnswer).toBeVisible();

      const secondQuestion = screen.getByText("How do I add a tenant?");
      fireEvent.click(secondQuestion);

      const secondAnswer = screen.getByText(
        "Go to the tenants page and click add tenant.",
      );
      expect(secondAnswer).toBeVisible();
    });
  });
});
