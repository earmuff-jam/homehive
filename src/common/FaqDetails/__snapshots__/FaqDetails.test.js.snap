// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-6rg9iv-MuiStack-root"
  >
    <div
      class="MuiStack-root css-5545du-MuiStack-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1aso48g-MuiTypography-root"
      >
        Frequently asked questions
      </h5>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-1ypmsg6-MuiTypography-root"
      >
        Answers to common questions you may have.
      </h6>
      <div
        class="MuiBox-root css-56sg73"
      />
    </div>
    <div
      class="MuiBox-root css-eaag1a"
    />
  </div>
</DocumentFragment>
`;
